src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalUsingVarRewriter.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.Lowering.LocalRewriter
+{
+    internal class LocalUsingVarRewriter : BoundTreeRewriterWithStackGuard
+    {
+        public static BoundNode Rewrite(BoundStatement statement)
+        {
+            var localUsingVarRewriter = new LocalUsingVarRewriter();
+            return localUsingVarRewriter.Visit(statement);
+        }
+
+        public override BoundNode VisitBlock(BoundBlock node)
```

Comment:
Now that I'm looking at the full change, I think having a dedicated rewriter may be overkill. This seems like it could fit into the existing LocalRewriter `VisitBlock` without too much extra complexity. I don't think we should try to do that as part of this change, but I've filed a bug to follow-up: https://github.com/dotnet/roslyn/issues/28707